Sources/StreamChat/Utils/ThreadSafeWeakCollection.swift

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+final class ThreadSafeWeakCollection<T: AnyObject> {
+    private let queue = DispatchQueue(label: "io.stream.com.weak-collection")
```

Comment:
This means that all `TheadSafeWeakCollection`'s will use the same queue. Shouldn't we pass the label in the initializer, so that each collection has its own queue?

Reply:
Nope, not at all. The label is mainly used for debugging purposes, DispatchQueue will not create unique queues based on it.

Reply:
You are right 👍 This a wrong assumption from me